### DISCUSSION TITLE
new addon idea (editor)

### DISCUSSION THREAD

**Discussion by niceEli** [ScratchAddons/ScratchAddons#3728] [Ideas] [answered]
javascipt

yes i said it javascript i think you should make a switch for the editor that allows you to do blocks or JavaScript (Maybe python)

(This was a dumb issue I made a while ago, pls don't respond, I already know scratch won't get JS support)

**Comment by Endermaster3690** [comment]
javascript to scratch blocks? scratch blocks to javascript? ok what?

**Reply by niceEli** [reply]
both (use microsoft makecode as example)

**Reply by Endermaster3690** [reply]
uhh ok???

**Reply by niceEli** [reply]
its a good idea for people who want a challenge to code

**Comment by kccuber-scratch** [comment, marked as answer]
You can meanwhile use [LeopardJS](https://leopardjs.com).

**Reply by GrahamSH-LLK** [reply]
Leapord runs projects in JS, afaik that's not what the requestor was asking

**Reply by kccuber-scratch** [reply]
This is what the requester was asking:
> allows you to do blocks or JavaScript

Well, if you look at the Leopard website theres a key to what blocks turn into what javascript. You can actually use this to change your projects directly in the leopard code sandbox.
